src/cache/precomputed_quantities.jl

```diff
@@ -334,33 +334,23 @@ function eddy_diffusivity_coefficient(
     Ri_a::FT,
     Ri_c::FT,
     κ::FT,
+    ᶜp::FT,
 ) where {FT}
     # Equations (17), (18)
-    if z < f_b * h
-        K_b =
-            compute_surface_layer_diffusivity(z, z₀, κ, C_E, Ri, Ri_a, Ri_c, uₐ)
+    K_b =
+        compute_surface_layer_diffusivity(z, z₀, κ, C_E, Ri, Ri_a, Ri_c, uₐ)
+    if z <= f_b * h
         return K_b
     elseif f_b * h < z < h
-        K_b = compute_surface_layer_diffusivity(
-            f_b * h,
-            z₀,
-            κ,
-            C_E,
-            Ri,
-            Ri_a,
-            Ri_c,
-            uₐ,
-        )
         K = K_b * (z / f_b / h) * (1 - (z - f_b * h) / (1 - f_b) / h)^2
         return K
     else
-        return FT(0)
+        return eddy_diffusivity_coefficient(C_E, uₐ, z, ᶜp)
     end
 end
 
 function compute_boundary_layer_height!(
```

Comment:
If possible, let's make the boundary layer height the first level where Ri exceeds Ri_c.